src/ORM/Association/HasMany.php

```diff
@@ -129,7 +129,7 @@ public function saveStrategy($strategy = null)
     public function saveAssociated(EntityInterface $entity, array $options = [])
     {
         $targetEntities = $entity->get($this->property());
-        if (empty($targetEntities)) {
+        if (empty($targetEntities) && $this->_saveStrategy !== self::SAVE_REPLACE) {
```

Comment:
Indeed. But when I wrote this line I was thinking in the possibility that, in the future, more save strategies could be implemented (I don't really know what more), so this line would be compliant, because it means: "It doesn't matter the save strategy you have, if it is not a replace, then there would be no reason why you would save an empty list of associated entities"